src/d3d9/d3d9_device.cpp

```diff
@@ -1093,6 +1095,12 @@ namespace dxvk {
     if (unlikely(iSwapChain != 0))
       return D3DERR_INVALIDCALL;
 
+    #ifdef _WIN32
+    if (!IsGamescopeWSIEnabled()) {
+      return D3D9SwapChainEx::GetFrontBufferDataGDI(pDestSurface);
```

Comment:
do we need to be robust against the GDI call failing (e.g. due to a non-rgba8 swapchain) or is this only supported for specific formats?

Reply:
That should work fine.

`GetFrontBufferData` requires the target surface to be `A8R8G8B8`, so that one isn't gonna be an issue.
On the GDI side, we're capturing the contents by blitting into a compatible bitmap which can have a device specific format. After that we're using `GetDIBits`. DI stands for Device Independent and that function handles the conversion to the specified format for us.

> The GetDIBits function retrieves the bits of the specified compatible bitmap and copies them into a buffer as a DIB using the specified format.